testsuite/src/test/java/org/jboss/da/test/server/ReportsWithWhitelistedArtifactsRemoteTest.java

```diff
@@ -0,0 +1,492 @@
+
+        // Remove products
+        productService.removeProduct("Product1", "1.0.0");
+        productService.removeProduct("Product1", "2.0.0");
+        productService.removeProduct("Product2", "1.0.0");
+    }
+
+    // This allows to test:
+    // - whitelist lookup by product
+    // - whitelist lookup by product version
+    // - whitelist lookup without either (get all - 3 additions to top level)
+    // - whitelist addition to top level
+    // - whitelist addition to top-1 level
+
+    private SCMReportRequest getDefaultSCMRequest() {
+        SCMLocator locator = new SCMLocator(
+                "https://github.com/project-ncl/dependency-analysis.git", "05a878", "common", null);
+
+        SCMReportRequest request = new SCMReportRequest();
+        request.setScml(locator);
+
+        return request;
+    }
+
+    @Test
+    public void testScmAllWhiteArtifactsPresent() throws ScmException, PomAnalysisException, CommunicationException{
+
+        SCMReportRequest request = getDefaultSCMRequest();
+
+        Optional<ArtifactReport> ar = reportGenerator.getReportFromSCM(request);
+        
+        // Ensure we got an artifact report back 
+        assertTrue(ar.isPresent());
+        ArtifactReport artifactReport = ar.get();
+    
+        // Ensure all dependencies from whitelist are present
+        assertTrue(artifactReport.getAvailableVersions().contains(whiteArtifact1_1_1.getVersion()));
+        assertTrue(artifactReport.getAvailableVersions().contains(whiteArtifact1_2_1.getVersion()));
+        assertTrue(artifactReport.getAvailableVersions().contains(whiteArtifact2_1_a.getVersion()));
+        
+        // These have different GA so shouldn't be there
+        assertFalse(artifactReport.getAvailableVersions().contains(whiteArtifact1_1_2.getVersion()));
+        assertFalse(artifactReport.getAvailableVersions().contains(whiteArtifact1_2_2.getVersion()));
+        assertFalse(artifactReport.getAvailableVersions().contains(whiteArtifact2_1_b.getVersion()));
+        assertFalse(artifactReport.getAvailableVersions().contains(whiteArtifact2_1_c.getVersion()));
+        
+        // Find jackson artifactReport and check it contains the jackson version
+        Optional<ArtifactReport> dependencyJacksonDatabind = artifactReport.getDependencies().stream()
+            .filter(x -> 
+                        x.getArtifactId().equals(whiteArtifact2_1_b.getArtifactId())
+                        &&
+                        x.getGroupId().equals(whiteArtifact2_1_b.getGroupId())
+                   )
+            .findFirst();
+            
+        assertTrue(dependencyJacksonDatabind.isPresent());
+        assertTrue(dependencyJacksonDatabind.get().getAvailableVersions().contains(whiteArtifact2_1_b.getVersion()));
+    }    @Test
+    public void testScmWhiteArtifactsPresentForDependenciesWithSingleProduct() throws ScmException, PomAnalysisException, CommunicationException{
```

Comment:
Tried to make it more readable.. the formatting checks seem to pass, so is there an issue?